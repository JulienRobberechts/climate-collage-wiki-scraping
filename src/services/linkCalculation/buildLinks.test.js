/**
 * @jest-environment node
 */
const { buildLinks, buildAllLinks } = require('./buildLinks');
const cards1and3FilePath = './src/tests/data/cards-1-3.json';
const { getObject } = require('../fileServices/readFile');

describe('buildLinks', () => {
  it('build link carte_1', async () => {
    const cardNumber = 1;
    const cards = await getObject(cards1and3FilePath);
    const linksInfo = buildLinks(cards, cardNumber);
    expect(linksInfo).toStrictEqual([{
      fromNum: 1,
      toNum: 2,
    },
    {
      fromNum: 1,
      toNum: 3,
    }]);
  });
  it('build link carte_2', async () => {
    const cardNumber = 2;
    const cards = await getObject(cards1and3FilePath);
    const linksInfo = buildLinks(cards, cardNumber);
    expect(linksInfo).toStrictEqual([{
      fromNum: 1,
      toNum: 2,
    },
    {
      fromNum: 2,
      toNum: 3,
    }]);
  });
});

describe('buildAllLinks', () => {
  it('build links', async () => {
    const cardNumber = 1;
    const cards = await getObject(cards1and3FilePath);
    const linksInfo = buildAllLinks(cards);
    expect(linksInfo).toMatchSnapshot();
  });
});
