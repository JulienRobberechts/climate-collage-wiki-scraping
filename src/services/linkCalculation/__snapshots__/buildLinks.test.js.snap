// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildAllLinks build links 1`] = `
Array [
  Object {
    "fromNum": 1,
    "toNum": 2,
  },
  Object {
    "fromNum": 1,
    "toNum": 3,
  },
  Object {
    "fromNum": 2,
    "toNum": 3,
  },
]
`;
